Framework/Built_In_Automation/Mobile/CrossPlatform/Appium/BuiltInFunctions.py

```diff
@@ -4606,4 +4617,173 @@ def scroll_to_element(data_set):
 
     except Exception:
         return CommonUtil.Exception_Handler(sys.exc_info(), None, "Error could not scroll the element")
+    
+
+def zoom_action(data_set):
+    """
+    This function is used for zoom in or zoom out. The scale and count parameters controls the amount of zoom.
+    :param data_set:
+        [['Action',         'zoom parameter',       'zoom in'/'zoom out'],
+        ['Scale',           'zoom parameter',        3],
+        ['Count',           'zoom parameter',        5],
+        ['zoom',            'appium action',        'Zoom in or out']]
+    """        
+    sModuleInfo = inspect.currentframe().f_code.co_name + " : " + MODULE_NAME
+
+    global appium_driver
+
+    zoom_element = LocateElement.Get_Element(data_set, appium_driver)
+    if zoom_element == "zeuz_failed":
+        CommonUtil.ExecLog(sModuleInfo, "Zoom element is not found", 3)
+        return "zeuz_failed"
+    
+    height = zoom_element.size["height"]
+    width = zoom_element.size["width"]
+    xstart_location = zoom_element.location["x"]
+    ystart_location = zoom_element.location["y"]
+    xx = xstart_location+int((width)//2)
+    yy = ystart_location+int((height)//2)
+
+    xs = xstart_location
+    xe = xstart_location+width  
+
+    action = ''
+    scale = 3
+    count = 5
+    const = 25
+    double_tap = False
+
+    xx1 = xx-const
+    xx2 = xx+const
+    x_cord1 = None
+    x_cord2 = None
+
+
+    try:
+        for left, mid, right in data_set:
+            if left.strip().lower() == 'action':
+                action = right.strip().lower().replace(' ','')
+            elif left.strip().lower() == 'scale':
+                scale = int(right)
+            elif left.strip().lower() == 'count':
+                count = int(right)
+            elif 'coordinates' in left.strip().lower():
+                cords = right.strip()
```

Comment:
have you tested this code? I dont think it will work

Reply:
No I haven't tested it. In fact, we can remove this part. The idea was the user could provide a (x,y) coordinate and the zoom/pan action would occur keeping that coordinate in the center. But now as we have added element parameter I think it's redundant to keep the option of the coordinates as well. What do you think?

Reply:
okay. since there is no client need as well, you can remove it

Reply:
Done